packages/plugins/c-sharp/c-sharp/src/visitor.ts

```diff
@@ -52,12 +56,150 @@ export interface CSharpResolverParsedConfig extends ParsedConfig {
   emitRecords: boolean;
   emitJsonAttributes: boolean;
   jsonAttributesSource: JsonAttributesSource;
+  emitCompositionTypes: boolean;
 }
 
 export class CSharpResolversVisitor extends BaseVisitor<CSharpResolversPluginRawConfig, CSharpResolverParsedConfig> {
   private readonly jsonAttributesConfiguration: JsonAttributesSourceConfiguration;
 
-  constructor(rawConfig: CSharpResolversPluginRawConfig, private _schema: GraphQLSchema) {
+  /// A map from types implementing unions, and the unions that they implement
+  private readonly compositionTypeToImplementationsMap: CompositionTypesMap;
+
+  /// A set of all types that are union types
+  private readonly compositionTypeNames: Set<string>;
+
+  private compositionTypeConverterTag = '[JsonConverter(typeof(CompositionTypeConverter))]';
+
+  private compositionTypeListConverterTag = '[JsonConverter(typeof(CompositionTypeListConverter))]';
+
+  private compositionTypeCacheDefinition = `
+/// <summary>
+/// A cache of generated JToken::ToObject[TargetType] for each __typeName
+/// </summary>
+private static ConcurrentDictionary<string, Func<JToken, object>> ToObjectForTypenameCache = new ConcurrentDictionary<string, Func<JToken, object>>();
+  `;
+
+  private compositionTypeConvertersBlock = `
+/// <summary>
+/// Given __typeName returns JToken::ToObject[__typeName]. (via cache to improve performance)
+/// </summary>
+/// <param name="typeName">The __typeName</param>
+/// <returns>JToken::ToObject[__typeName]</returns>
+public static Func<JToken, object> GetToObjectMethodForTargetType(string typeName)
+{
+  if (!ToObjectForTypenameCache.ContainsKey(typeName))
+  {
+    // Get the type corresponding to the typename
+    Type targetType = typeof(YammerGQLTypes).Assembly
```

Comment:
Is YammerGQLTypes what you want here? I was applying your change against master to create a local patch and it's breaking here.

Reply:
There might be more correct ways of doing this but I just stumbled across this and found this worked for me in a pinch:

```
diff --git a/packages/plugins/c-sharp/c-sharp/src/visitor.ts b/packages/plugins/c-sharp/c-sharp/src/visitor.ts
index ad6af2859..9590d289d 100644
--- a/packages/plugins/c-sharp/c-sharp/src/visitor.ts
+++ b/packages/plugins/c-sharp/c-sharp/src/visitor.ts
@@ -79,7 +79,7 @@ export class CSharpResolversVisitor extends BaseVisitor<CSharpResolversPluginRaw
 private static ConcurrentDictionary<string, Func<JToken, object>> ToObjectForTypenameCache = new ConcurrentDictionary<string, Func<JToken, object>>();
   `;
 
-  private compositionTypeConvertersBlock = `
+  private compositionTypeConvertersBlock = (namespace: string, className: string) => `
 /// <summary>
 /// Given __typeName returns JToken::ToObject[__typeName]. (via cache to improve performance)
 /// </summary>
@@ -90,7 +90,7 @@ public static Func<JToken, object> GetToObjectMethodForTargetType(string typeNam
   if (!ToObjectForTypenameCache.ContainsKey(typeName))
   {
     // Get the type corresponding to the typename
-    Type targetType = typeof(YammerGQLTypes).Assembly
+    Type targetType = typeof(${namespace}.${className}).Assembly
       .GetTypes()
       .ToList()
       .Where(t => t.Name == typeName)
@@ -262,7 +262,7 @@ public class CompositionTypeListConverter : JsonConverter
 
   public addCompositionTypeConverterDefinitions(blocks: string[]): string[] {
     return this._parsedConfig.emitCompositionTypes
-      ? [this.compositionTypeCacheDefinition, ...blocks, this.compositionTypeConvertersBlock]
+      ? [this.compositionTypeCacheDefinition, ...blocks, this.compositionTypeConvertersBlock(this.config.namespaceName, this.config.className)]
       : blocks;
   }
 

```

Reply:
Thanks for spotting this, @jharwood0 patch seems to make sense, I will test it out myself and push another iteration!

Reply:
@cicciodm, I'm staying at your disposal on The Guild side to help you make this PR move forward!